### PR TITLE
Fix code analysis warning in a test utility method

### PR DESCRIPTION
**Pull request by kartheekp-ms**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2027

Regression? Last working version:

## Description

Please refer to the Az DO work item linked in the tracking bug for more details.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] N/A 

- **Documentation**
  - [x] N/A

### CODE REVIEW

File: test/TestUtilities/Test.Utility/ZipExtensions.cs
Comment (zivkan) [inline test/TestUtilities/Test.Utility/ZipExtensions.cs]: You can do this (calculate the `fullDestDirPath`) outside the `foreach` loop to minimize string allocations.

Comment (kartheekp-ms) [reply test/TestUtilities/Test.Utility/ZipExtensions.cs]: Good catch. Fixed in the latest commit.

**Review by zivkan — DISMISSED** [review]

**Review by erdembayar — DISMISSED** [review]

**Review by nkolev92 — DISMISSED** [review]

**Comment by kartheekp-ms** [comment]
CI checks passed. I will merge this PR after getting an approval.

**Review by erdembayar — APPROVED** [review]

**Pull request merged**
